LanguageFeatures/Class-modifiers/basic_restrictions_A04_t23.dart

```diff
@@ -0,0 +1,59 @@
+// Copyright (c) 2023, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+/// @assertion It is a compile-time error if:
+/// ...
+/// - A declaration has a base or final superdeclaration, and is not itself
+///   marked base, final or sealed.
+///
+/// @description Check that it is a compile-time error if a declaration is not
+/// `base`, `final` or `sealed` and has a superdeclaration marked `final`. Test
```

Comment:
If I understood the spec correctly, a declaration that has a proper superdeclaration marked `final` is a compile-time error regardless of the declaration itself having `base`, `final`, or `sealed` modifiers. Some exceptions may apply, based on the version of the libraries and them being core libraries.

Reply:
`final` class can be extended by `final`, `base` or `sealed` classes defined in the same library. One of the tests that checks it (and passes) is https://github.com/dart-lang/co19/blob/master/LanguageFeatures/Class-modifiers/basic_restrictions_A02_t07.dart

Reply:
Got it. Thanks for the clarification!